### ISSUE TITLE
App needed: calculator

### ISSUE THREAD

**Issue by CRImier** [ZeroPhone/ZPUI#100]
With Python's `eval` function, it should be easy to make a simple calculator... Well, until it comes to dealing with floating point numbers =D

```
root@zerophone-prototype:~/ZPUI# python
Python 2.7.13 (default, Nov 24 2017, 17:33:09)
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> 0.01/0.1
0.09999999999999999
>>> 0.1*0.1
0.010000000000000002
```

Having said that, we could really use a calculator in the default ZPUI install. Anybody up for making one?

**Comment by rud01f** [comment]
the problem is in how floating point numbers are internally stored: as a binary sequence and exponent. 0.1 decimal in binary is 0.00011001100110011.. etc., so it's not possible to store it perfectly. One idea is to use fixed point arithmetic - pre-multiply every number by let's say 1,000,000,000 , then divide the result.

**Comment by CRImier** [comment]
Yeah, I'm sure there's even a tried and true algorithm on Stackoverflow or somewhere else that most of the calculator apps use to avoid the precision problems =) The biggest effort will, surely, go into the calculator UI and user-friendliness, the `eval` part is mentioned in a way that "we need somebody to write an app, and in case you're worried about the backend you should use, here's an easy-to-use backend that we're OK with you using, and here are some amusing examples of it failing =)"

**Comment by j340m3** [comment]
Why not use [Decimals](https://docs.python.org/3/library/decimal.html)?

**Comment by CRImier** [comment]
@j340m3 this seems like exactly what we'd be looking for, from the first glance, thank you!

**Comment by CRImier** [comment]
@KyleMoran138 has made a basic app that uses `Decimals`, and it's now in `devel`. However, the UI isn't user-friendly enough - my proposal is to have a single-screen app, with this displayed:

- 

Later on, we could hide the controls' help, for a less bloated UI
